rest_framework_simplejwt/authentication.py

```diff
@@ -1,16 +1,26 @@
+from typing import Dict, List, Optional, Set, Tuple, Union
+
 from django.contrib.auth import get_user_model
+from django.contrib.auth.models import AbstractBaseUser, AbstractUser, AnonymousUser
 from django.utils.translation import gettext_lazy as _
 from rest_framework import HTTP_HEADER_ENCODING, authentication
+from rest_framework.request import Request
 
 from .exceptions import AuthenticationFailed, InvalidToken, TokenError
+from .models import TokenUser
 from .settings import api_settings
+from .tokens import Token
 
 AUTH_HEADER_TYPES = api_settings.AUTH_HEADER_TYPES
 
 if not isinstance(api_settings.AUTH_HEADER_TYPES, (list, tuple)):
     AUTH_HEADER_TYPES = (AUTH_HEADER_TYPES,)
 
-AUTH_HEADER_TYPE_BYTES = {h.encode(HTTP_HEADER_ENCODING) for h in AUTH_HEADER_TYPES}
+AUTH_HEADER_TYPE_BYTES: Set[bytes] = {
+    h.encode(HTTP_HEADER_ENCODING) for h in AUTH_HEADER_TYPES
+}
+
+AuthUser = Union[AbstractUser, AbstractBaseUser, AnonymousUser, TokenUser]
```

Comment:
Having doubts about this, whether it's correct and whether we would want to be implemented in this way.

Also considering using a TypeVar for this, or using a generic type without specifying the different user classs.